src/sctools/bam.py

```diff
@@ -187,20 +189,28 @@ def _cleanup(files_to_counts, files_to_names, rm_all=False):
         # cycler over files to assign new barcodes to next file
         file_cycler = cycle(files_to_counts.keys())
 
-        # create an empty map for barcodes to files
+        # create an empty map for (tag, barcode) to files
         tags_to_files = {}
 
-        # loop over input, partitioning barcodes into files
+        # loop over input; check each tag in sequence and partition barcodes into files
         for alignment in input_alignments:
-            try:
-                tag_content = alignment.get_tag(tag)
-            except KeyError:
+            for tag in tags:
```

Comment:
This seems expensive to me, is this "check each tag in sequence and partition" avoidable? (Potentially we could write this into a `map()` function so in the future multiprocessing could be added here for efficiency-sensible tasks, ignore me if I'm wrong here)

Reply:
You found a bug! It needed a `break` in the `try` block.